Text/Pandoc/Definition.hs

```diff
@@ -209,31 +211,91 @@ instance Eq Format where
+type RowSpan = Int
+
+-- | The number of positions in a column occupied by a cell; the
+-- height of a cell.
+type ColSpan = Int
+
 -- | Block element.
 data Block
-    = Plain [Inline]        -- ^ Plain text, not a paragraph
-    | Para [Inline]         -- ^ Paragraph
-    | LineBlock [[Inline]]  -- ^ Multiple non-breaking lines
-    | CodeBlock Attr Text -- ^ Code block (literal) with attributes
-    | RawBlock Format Text -- ^ Raw block
-    | BlockQuote [Block]    -- ^ Block quote (list of blocks)
-    | OrderedList ListAttributes [[Block]] -- ^ Ordered list (attributes
-                            -- and a list of items, each a list of blocks)
-    | BulletList [[Block]]  -- ^ Bullet list (list of items, each
-                            -- a list of blocks)
-    | DefinitionList [([Inline],[[Block]])]  -- ^ Definition list
-                            -- Each list item is a pair consisting of a
-                            -- term (a list of inlines) and one or more
-                            -- definitions (each a list of blocks)
-    | Header Int Attr [Inline] -- ^ Header - level (integer) and text (inlines)
-    | HorizontalRule        -- ^ Horizontal rule
-    | Table [Inline] [Alignment] [Double] [TableCell] [[TableCell]]  -- ^ Table,
-                            -- with caption, column alignments (required),
-                            -- relative column widths (0 = default),
-                            -- column headers (each a list of blocks), and
-                            -- rows (each a list of lists of blocks)
-    | Div Attr [Block]      -- ^ Generic block container with attributes
-    | Null                  -- ^ Nothing
+    -- | Plain text, not a paragraph
+    = Plain [Inline]
+    -- | Paragraph
+    | Para [Inline]
+    -- | Multiple non-breaking lines
+    | LineBlock [[Inline]]
+    -- | Code block (literal) with attributes
+    | CodeBlock Attr Text
+    -- | Raw block
+    | RawBlock Format Text
+    -- | Block quote (list of blocks)
+    | BlockQuote [Block]
+    -- | Ordered list (attributes and a list of items, each a list of
+    -- blocks)
+    | OrderedList ListAttributes [[Block]]
+    -- | Bullet list (list of items, each a list of blocks)
+    | BulletList [[Block]]
+    -- | Definition list. Each list item is a pair consisting of a
+    -- term (a list of inlines) and one or more definitions (each a
+    -- list of blocks)
+    | DefinitionList [([Inline],[[Block]])]
+    -- | Header - level (integer) and text (inlines)
+    | Header Int Attr [Inline]
+    -- | Horizontal rule
+    | HorizontalRule
+    -- | Table, with attributes, caption, optional short caption,
+    -- column alignments and widths (required), table head, table
+    -- body, and table foot
+    | Table Attr Caption [(Alignment, ColWidth)] TableHead TableBody TableFoot
+    -- | Generic block container with attributes
+    | Div Attr [Block]
+    -- | Nothing
+    | Null
     deriving (Eq, Ord, Read, Show, Typeable, Data, Generic)
 
```

Comment:
Can you really do the haddocks this way? I didn't know this form could be used for constructors.

Reply:
It's documented [here](https://haskell-haddock.readthedocs.io/en/latest/markup.html#constructors-and-record-fields). The generated haddocks seem to be fine, in any case. I changed it because the documentation source for `Block` was getting a little hard to read in the other style.

Reply:
Yeah, the `|` in the haddocks is too easily confused with the `|` in the type definition.

Reply:
Would you like me to revert the change?

Reply:
I'm okay with this style, I guess both approaches have their drawbacks.
In a source code editor syntax highlighting will help avoid confusion, and in the generated haddocks it will be the same either awy.